src/components/text-editor/examples/text-editor-height.tsx

```diff
@@ -0,0 +1,42 @@
+import { Component, h } from '@stencil/core';
+/**
+ * Height of the editor
+ *
+ * The text editor automatically adjusts its own height to fit the content inside.
+ * So as the user types, the editor will grow taller, potentially resizing its own
+ * container element.
+ *
+ * However, you can constrain the text editor to never grow beyond a certain height,
+ * using `--text-editor-max-height` property.
+ *
+ * In this example, the maximum height is set to `15rem`, which means that:
+ * 1. the editor will adjust itself to the content inside,
+ * pushing out its container and making it taller, until it reaches `15rem`.
+ * 1. and also the editor will try to fill the available surface area
+ * when you manually resize the container,
+ * (or some other element causes the container to grow taller), until it reaches `15rem`.
+ *
+ * :::important
+ * Naturally, if the container has a fixed small height,
+ * The editor which has a lot of content will grow taller than the container.
+ * Depending on the `overflow` property of the container,
+ * you may get undesired rendering, or parts of the content
+ * can become invisible for the end-users.
+ *
+ * You can test this by resizing this example (which will give it a hardcoded
+ * `width` and `height`), and then typing several lines of text!
+ *
+ * Therefore, you need to make sure that `--text-editor-max-height`
+ * is set to a value which correlates with the hardcoded height of the container.
+ * :::
```

Comment:
I tried really hard to force the editor to inherit its height from its parent, and never grow larger than its parent. However, it was a huge 😖  in the 🍑  and I couldn’t make it behave that way. There are too many nested containers and many ifs and buts. So the best way I could come up with is allowing the consumer to set a max-height in case they want to stop the editor from resizing after a certain point.

I think it’s a fair solution, but it should also be tested in the product, to make sure we get the expected behavior, for instance when the container is a grid child, or flex child, or has overflow, etc….

Reply:
Don't approve! I might have solved this